File: 1- Array - String/C#/9- String Compression.cs
public class Solution {
    public int Compress(char[] chars) {
        int n = chars.Length ; 
        if (n == 1)
            return 1 ; 

        int i = 0 ; 
        int ans = 0 ; 
        int j = 0 ; 
        while(i < n){
            char curr = chars[i] ; 
            int cnt = 0 ; 

            while(i < n && curr == chars[i]){
                i++ ; 
                cnt++ ; 
            }

            chars[j] = curr ; 
            j++ ; 

            if (cnt > 1){
                string s = cnt.ToString() ; 
                foreach(char c in s){
                    chars[j] = c ; 
                    j++ ; 
                }
            }
        }
        return j ; 
    }
}

File: 01- Array - String/C++/9- String Compression.cpp
class Solution {
    public:
        int compress(vector<char>& chars) {
            map<char , int> freq ; 
    
            for (auto c : chars)
                freq[c]++ ;
            string str = "" ; 
            for (auto [c , value] : freq) {
                if (freq[c] == 1)
                    str += c ;
                else if (freq[c] > 1) {
                    str += c ;
                    str += to_string(value);
                }
            }
    
            return str.size();
        }
    };